core/server/common/src/main/java/alluxio/master/journal/checkpoint/LongsCheckpointFormat.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal.checkpoint;
+
+import com.google.common.base.Preconditions;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Optional;
+
+/**
+ * Format for checkpoints consisting of a list of longs written by a DataOutputStream.
+ */
+public class LongsCheckpointFormat implements CheckpointFormat {
+  @Override
+  public LongsCheckpointReader createReader(CheckpointInputStream in) {
+    return new LongsCheckpointReader(in);
+  }
+
+  @Override
+  public void parseToHumanReadable(CheckpointInputStream in, PrintStream out) throws IOException {
+    LongsCheckpointReader reader = createReader(in);
+    Optional<Long> longOpt;
+    while ((longOpt = reader.nextLong()).isPresent()) {
+      out.printf("%d ", longOpt.get());
```

Comment:
New line vs space delimited, new line might be better from opening the file perspective?

Reply:
good call, switched to newline